docs/blog/microbit/2021-release.md

```diff
@@ -0,0 +1,84 @@
+
+## Data logging extension
+
+The new version of the micro:bit it is able to collect and store its own data! There is a new extension for MakeCode too that allows you to collect data from the micro:bit sensors and record the data on the board. This is helpful if you are running an experiment over a longer period of time or when the micro:bit is disconnected from a computer that does the data recording. It’s still in Beta, but here’s a short video demonstration:
+
+https://youtu.be/fWfwb8ZSsjc
+
+## Functions with array parameters
+
+Every year we add more functionality to our **Functions**. This year, we did a small update as a result of your feedback to support array types as parameters.
+
+The Function Editor now lets you select an array parameter:
+
+![Edit function](/static/blog/microbit/2021-release/edit-function.png)
+
+The example function here is using the array parameter:
+
+![Function sample](/static/blog/microbit/2021-release/function-sample.png)
+
+## Updates to Loops
+
+When you open up the **Loops Toolbox** drawer, you may see a couple of changes.
+
+### While loop
+
+We’ve found that "While True" loops can get students into a lot of trouble, often becoming infinite loops in their programs! So, we changed the default value to be "false" as a safer way for them to start using this block and to have the appropriate conditions set for exiting the loop.
+
+![New while loop block](/static/blog/microbit/2021-release/while.png)
+
+### Every time interval block
+
+This new block is useful for looping on a given time interval (in milliseconds). It's like the Forever loop but the code inside doesn't run again until an interval of time set for the loop expires first.
+
+![New every loop block](/static/blog/microbit/2021-release/every.png)
+
+## Text coding not available in Windows app
+
+Now that one-click downloads are supported with the web version of MakeCode, this release starts the journey to deprecate the Windows 10 app. You can no longer use the text editor in the app, and we will stop updating the app by the 2022 release next June.
+
+![No text editing message](/static/blog/microbit/2021-release/no-text-editing.png)
+
+If you do need an app with full capabilities but for offline use, we recommend using the [Offline App](https://makecode.microbit.org/offline).
+
+[![MakeCode App logo](/static/blog/microbit/2021-release/app-logo.png)](https://makecode.microbit.org/offline)
+
+Thank you to everyone who has logged bugs, helped with translations, and suggested new features throughout the year! Any new bugs you find in this release, please log them in [GitHub](https://github.com/Microsoft/pxt-microbit/issues).
+
+Any other comments, suggestions, and feedback, please participate in the micro:bit community on [Slack](https://tech.microbit.org/get-involved/where-to-find/). And, please let us know what you think of this year’s updates on Twitter [@MSMakeCode](https://twitter.com/MSMakeCode).
```

Comment:
This Slack link appears to be bad. Did you want https://tech.microbit.org/community/?

Reply:
Ah, yes please!  Thanks for catching that!